airflow/providers/jenkins/hooks/jenkins.py

```diff
@@ -29,7 +30,7 @@ class JenkinsHook(BaseHook):
     Hook to manage connection to jenkins server
     """
 
-    def __init__(self, conn_id='jenkins_default'):
+    def __init__(self, conn_id: Union[str, Text] = 'jenkins_default') -> None:
```

Comment:
```suggestion
    def __init__(self, conn_id: str = 'jenkins_default') -> None:
```
That's the usual type we use for connection id